extension/app/src/components/conversation/AttachFragment.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  AttachmentIcon,
+  Button,
+  DocumentIcon,
+  DocumentTextIcon,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  ImageIcon,
+} from "@dust-tt/sparkle";
+import { supportedFileExtensions } from "@dust-tt/types";
+import type { EditorService } from "@extension/components/input_bar/editor/useCustomEditor";
+import type { FileUploaderService } from "@extension/hooks/useFileUploaderService";
+import { useRef } from "react";
+
+type AttachFragmentProps = {
+  fileUploaderService: FileUploaderService;
+  editorService: EditorService;
+};
+
+export const AttachFragment = ({
+  fileUploaderService,
+  editorService,
+}: AttachFragmentProps) => {
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  return (
+    <>
+      <input
+        accept={supportedFileExtensions.join(",")}
+        onChange={async (e) => {
+          await fileUploaderService.handleFileChange(e);
+          if (fileInputRef.current) {
+            fileInputRef.current.value = "";
+          }
+          editorService.focusEnd();
+        }}
+        ref={fileInputRef}
+        style={{ display: "none" }}
+        type="file"
+        multiple={true}
+      />
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button
+            icon={AttachmentIcon}
+            variant="ghost"
+            isSelect
+            size="xs"
+            tooltip="Pick an assistant"
+          />
+        </DropdownMenuTrigger>
+        <DropdownMenuContent className="min-w-[300px]">
+          <DropdownMenuItem
+            icon={DocumentIcon}
+            label={"Attach file"}
+            onClick={async () => {
+              fileInputRef.current?.click();
+            }}
+          />
+          <DropdownMenuItem
```

Comment:
This is a great idea yet worth discussing quickly with Alex (if not already done of course) to see if we should reconsider the styling of the "Tab sharing" button. This is the equivalent but as a one shot.

Reply:
👍  will check with him